client/src/app/auth/firebase-auth.service.ts

```diff
@@ -19,10 +19,10 @@ export class FirebaseAuthService extends AuthService {
     // if it's the very first time. If `user` is null, we make sure to sign in anonymously
     // so that a user is always logged-in in some way.
     //
-    // Use this method to subscribe to any auth changes througout the lifecycle of
+    // Use this method to subscribe to any auth changes throughout the lifecycle of
     // a user session.
     return (
-      new Observable(obs => firebase.auth().onAuthStateChanged(obs))
+      new Observable(obs => firebase.auth().onIdTokenChanged(obs))
```

Comment:
Really cool! Didn't know about `onIdTokenChanged()`. It seems it covers everything `onAuthStateChanged()` covers, plus token refresh events. 

Would you mind updating the surrounding comment as well? It still says `onAuthStateChanged()`

Reply:
Done!